tests/test_Executor.cc

```diff
@@ -151,6 +151,50 @@ TEST_CASE("Executor: Run a failing sequence asynchronously", "[Executor]")
     REQUIRE(sequence.get_error_message() != "");
 }
 
+TEST_CASE("Executor: cancel() endless step loop", "[Executor]")
+{
+    Context context;
+    context.log_error_function = nullptr; // suppress error output on console
+
+    Sequence sequence{ "test_sequence" };
+    sequence.push_back(Step{ Step::type_while }.set_script("return true"));
+    sequence.push_back(Step{ Step::type_action }.set_script("a = 1"));
+    sequence.push_back(Step{ Step::type_end });
+
+    Executor executor;
+
+    executor.run_asynchronously(sequence, context);
+
+    gul14::sleep(1ms);
```

Comment:
The timing of the test is still brittle and some test runs failed while others did not. Make sure to run the tests multiple times ;-D